homework/index.js

```diff
@@ -30,14 +30,120 @@
     return elem;
   }
 
+  function clearContainer(container) {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function renderError(err) {
+    const listContainer = document.getElementById('list-container');
+    clearContainer(listContainer);
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert alert-error' });
+  }
+
+  function addRow(tbody, label, repository) {
```

Comment:
Excellent that you tried to make your code DRY by adding this function. It is a good first try, however, in your implementation, you do not gain much as it needs detailed knowledge about the structure of the repository object. Alternatively, the `addRow` function can be as simple as this:

```js
function addRow(tbody, label, value = '') {
  const row = createAndAppend('tr', tbody);
  createAndAppend('td', row, { text: `${label} :`, class: 'label' });
  createAndAppend('td', row, { text: value });
  return row;
}
```

In this implementation, the `addRow` function is completely detached from any knowledge about where the data comes from. It just needs a parent `tbody` element and two string arguments, one for the label text and one for the text value to be displayed. You can call use it like this (note how the first row get some extra treatment to add the required `<a>` tag):

```js
const firstRow = addRow(tbody, 'Repository');
createAndAppend('a', firstRow.lastChild, {
  href: repo.html_url,
  target: '_blank',
  text: repo.name,
});

addRow(tbody, 'Description', repo.description);
addRow(tbody, 'Forks', repo.forks);
addRow(tbody, 'Updated', new Date(repo.updated_at).toLocaleString());
```

This `addRow` function is reusable whereas your version is highly specialized to work only with repository objects.